### PR TITLE
[#2302][FOLLOWUP] dev: Fix the issue of disk space

### PR DESCRIPTION
**Pull request by qqqttt123**
### What changes were proposed in this pull request?
Remove unused packages and then free enough disk space.

### Why are the changes needed?
If we don't have this pr, we will encounter the error.
https://github.com/datastrato/gravitino/actions/runs/8076657520

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
I have tested it by using https://github.com/datastrato/gravitino/actions/runs/8077266349/job/22067221802

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -81,4 +81,4 @@ jobs:
           distribution: 'temurin'
 
       - name: Build and Push the Docker image
-        run: ./dev/docker/build-docker.sh --platform all --type ${image_type} --image ${image_name} --tag ${{ github.event.inputs.tag }} --latest
+        run: sudo rm -rf /usr/local/lib/android && sudo rm -rf /opt/hostedtoolcache/CodeQL && ./dev/docker/build-docker.sh --platform all --type ${image_type} --image ${image_name} --tag ${{ github.event.inputs.tag }} --latest
```
Comment (yuqi1129) [inline .github/workflows/docker-image.yml:84]: What does the directory `/opt/hostedtoolcache/CodeQL` is used for?

Comment (jerqi) [reply .github/workflows/docker-image.yml:84]: Git action vm pre-installed some packages. We don't need it. So we can remove it. We don't need to care what is used for . Arrow pipeline also removes this package.

Comment (qqqttt123) [reply .github/workflows/docker-image.yml:84]: Git action vm pre-installed some packages. We don't need it. So we can remove it. We don't need to care what is used for . Arrow pipeline also removes this package.

Comment (yuqi1129) [reply .github/workflows/docker-image.yml:84]: I see.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**